src/sentry/api/serializers/models/organization_member/base.py

```diff
@@ -96,17 +99,31 @@ def serialize(
         user: User | RpcUser | AnonymousUser,
         **kwargs: Any,
     ) -> OrganizationMemberResponse:
+        serialized_user = attrs["user"]
+        if obj.user_id:
+            # Only use the user's primary email from the serialized user data
+            email = serialized_user["email"] if serialized_user else obj.email
+        else:
+            # For invited members, use the invitation email
+            email = obj.email
+
+        # helping mypy - we know email will never be None based on the model
+        assert email is not None
+        email = str(email)
```

Comment:
I think because of the `assert` you should be able to drop the two `str(...)` calls below

but also it appears `OrganizationMember` has nullable `email` field so I'm not certain this assertion will pass?

Reply:
Removed the `str(...)` calls.

I also added comments to clarify the reasoning around the asserts. It basically boils down to two cases:

1. An actual user that is a part of the organization will have a null value in `email` of their corresponding `OrganizationMember` row. The `user_id` will be populated.
2. A user that has been invited, but has not accepted the invite will not have a `user_id` (because they do not yet have a full account), but will have a value in `email`. 

https://github.com/getsentry/sentry/blob/b4ac07868121fb7909ac29c75dfd1bd5875c923c/src/sentry/api/serializers/models/organization_member/base.py#L102-L111